### DISCUSSION TITLE
Extended Functionality for API Tokens

### DISCUSSION THREAD

**Discussion by cushdog** [PrairieLearn/PrairieLearn#10604] [Ideas]
Hi!

I apologize if this has already been discussed, but I'm working on a larger app that aims at improving the student experience, and I wanted to integrate PL data, and I noticed that, if I wanted to access any course data of a course that I'm currently enrolled in, with a personal access token that I'm not actually able to, I can only access courses that I'm an instructor of. I'm curious to know if there has been any talk about changing this, and if there has, I might be interested in creating a pull request for this myself (I'm newer to larger codebases like this and it'd be much appreciated if someone could point me in the general right direction, or if I'm able to add my own OAuth flow to my app that handles this then some help with that would be greatly appreciated!).

Thanks!

**Comment by nwalters512** [comment]
👋 This was previously requested and briefly discussed here: https://github.com/PrairieLearn/PrairieLearn/issues/5312. Let's move any further discussion about implementation details to that issue.
